airflow/models/baseoperator.py

```diff
@@ -1107,9 +1107,11 @@ def dag(self, dag: DAG | None):
 
         if self.__from_mapped:
             pass  # Don't add to DAG -- the mapped task takes the place.
-        elif self.task_id not in dag.task_dict:
-            dag.add_task(self)
-        elif self.task_id in dag.task_dict and dag.task_dict[self.task_id] is not self:
+        elif (
+            self.task_id not in dag.task_dict
+            or self.task_id in dag.task_dict
+            and dag.task_dict[self.task_id] is not self
+        ):
```

Comment:
```suggestion
        elif (
            self.task_id not in dag.task_dict
            or dag.task_dict[self.task_id] is not self
        ):
```

or just

```suggestion
        elif dag.task_dict.get(self.task_id) is not self:
```

Reply:
Indeed, the second suggestion should do the job.